### DISCUSSION TITLE
iTwin Exporter for Datasmith Crash Problem

### DISCUSSION THREAD

**Discussion by tahaerenoglu** [iTwin/itwinjs-core#5370] [Q&A]
iTwin Exporter for Datasmith stops working and shuts down by itself after clicking export option. I tried by two different models but the problem was still there.

**Comment by MichaelBelousov** [comment]
@tahaerenoglu did these models export previously? Can you confirm the version you have from the debug log available from the `(?)` icon in the top right?

**Reply by tahaerenoglu** [reply]
No they didn't exported previously. And the debug log information mentioned below.

Info    |Logger| 'initialized logger'; {"productVersion":"0.8.2"}
Info    |imodeljs-addon| 'Initialized iModel.js addon to region 'PROD'.'; 
Info    |itwin-client.Request| 'https://ims.bentley.com/connect/userinfo'; 
Info    |imodeljs-addon| 'Initialized iModel.js addon to region 'PROD'.'; 
Info    |iTwinDatasmithExporter.Backend| 'checking for updates...'; 
Info    |iTwinDatasmithExporter.Backend| 'no updates found.'; 
Info    |itwin-client.Request| 'https://ims.bentley.com/connect/userinfo';

**Reply by tahaerenoglu** [reply]
![Test](https://user-images.githubusercontent.com/130343823/231104797-534ea337-6d51-4f07-b197-f906d7a2b1b3.png)

It stucks at this stage and crashes without any error or warning.

**Reply by MichaelBelousov** [reply]
Does it crash immediately or after a prolonged wait after exporting?

**Comment by tahaerenoglu** [comment]
İt crashes immediately. Shuts down by itself

**Reply by MichaelBelousov** [reply]
@tahaerenoglu is there an email at which I can reach you to ask some questions about your data?

**Reply by tahaerenoglu** [reply]
You can see details on my profile

**Comment by MichaelBelousov** [comment]
@tahaerenoglu I have identified the issue, it is related to us not handling special characters in some cases when exporting meshes for datasmith to files in the `_Assets` folder. I have a fix where errors in this part of the application will show to the user instead of causing crashes, and I am changing it to ignore those characters when generating the mesh paths, so your export should work in the next version. I will reply here when that version is released.

**Reply by MichaelBelousov** [reply]
on another note, this may be unique to the datasmith SDK's handling of file paths.
